test-fork/vault/savePlus.spec.ts

```diff
@@ -1500,7 +1541,11 @@ describe("Save+ Basic and Meta Vaults", async () => {
                 it("total redeem", async () => {
                     await assertVaultRedeem(daiWhale, daiToken, daiMetaVault, dataEmitter)
                     await assertVaultRedeem(usdcWhale, usdcToken, usdcMetaVault, dataEmitter)
-                    await assertVaultRedeem(usdtWhale, usdtToken, usdtMetaVault, dataEmitter)
+                    const redeemAmount = await usdtMetaVault.connect(usdtWhale.signer).maxRedeem(usdtWhale.address)
+                    // TODO , when the last user of the vault redeems all USDT, the pool has issues, available assets < maxWithdraw/Redeem.
+                    // some slippage tolerance can be added to solve this issue instead of reverting the tx.
+                    // @contracts/vault/allocate/PeriodicAllocationAbstractVault.sol#270L `if (totalUnderlyingAssets >= requiredAssets)`
```

Comment:
@naddison36  what do you think?  the difference is less than  2 decimal points but that is the root cause of the random issues we were having

Reply:
This is definitely a problem. `previewRedeem` needs to return less than or equal to the assets returned from `redeem`.
I'll see if I can find out why